lib/vandelay/integrations/vendor/base.rb

```diff
@@ -0,0 +1,42 @@
+require 'net/http'
+
+module Vandelay
+    module Integrations
+      module Vendor
+        class Base
+            def initialize
+                @token_data = nil
+                @patient_data = nil
+            end
+
+            def authenticate(endpoint)
+                response = get(endpoint)
+                @token_data = JSON(response)
+            end
+
+            def patient(endpoint, token)
+                response = get(endpoint, token)
+                @patient_data = JSON(response)
+            end
```

Comment:
* I added `patient` to fetch the entire patient object from the external api for further processing.
* The idea behind the base class is that it acts like a wrapper for the external API and thus responsible for all the needed api implementations.
* `patient_record` is to be implemented in the child classes because the data payload structure for each child is different and thus need to be handled in their respective class.